### ISSUE TITLE
model weights and inference

### ISSUE THREAD

**Issue by kvnsng** [Mr81wx/HoopTransformer#1]
Hello - thanks for this very interesting paper and work! 

I was just wondering if you had model weights and inference code we can try. Thank you!

**Comment by Mr81wx** [comment]
loading_ckpt
<https://mail.google.com/mail/u/0?ui=2&ik=96449053f0&attid=0.1&permmsgid=msg-a:r-7239896480587092773&view=att&disp=safe&realattid=f_m1anbuxg0>
(17 K)
<https://mail.google.com/mail/u/0?ui=2&ik=96449053f0&attid=0.1&permmsgid=msg-a:r-7239896480587092773&view=att&disp=safe&realattid=f_m1anbuxg0>Hi
Kevin,

Sorry for the late reply.

I attached the checkpoints for our model. You can load the parameters under
Pytorch Lightning's mode.

However, there are strict limitations on the input format of the model, you
can refer to the attached file for the create_batch_class function, this is
used for classification finetuning.

Best

Xing
 Checkpoints_C.ckpt
<https://drive.google.com/open?id=1qdcXb9CRJZgniNVaViiZaQXNiWxpiTnQ>

Kevin Song ***@***.***> 于2024年9月12日周四 20:22写道：

**Comment by kvnsng** [comment]
Thank you for the access! I don't think I have the file with `create_batch_class` function. Would you be able to add that to the github repo? If not, could you send it to my email? songkevinc@gmail.com

**Comment by Mr81wx** [comment]
try this link
https://colab.research.google.com/drive/1YCGmMn5CBC0j8d5sj45Y53FcAvxvoJWE?usp=drive_link


Kevin Song ***@***.***> 于2024年9月20日周五 15:13写道：

**Comment by kvnsng** [comment]
That works. Thank you and I'll go ahead and close the issue!

**Comment by Mr81wx** [comment]
Great!

We have updated the new model extensions for defense players. Please let me know if you need them.

We're also looking for some soccer and football data to see this framework's performance. If you have similar resources and ideas, we can work together!

**Comment by kvnsng** [comment]
Sounds good. thank you!

**Comment by kvnsng** [comment]
Another quick question on the play embeddings... I noticed the output from the encoder is in the shape of [6,121,256]. When you create play embeddings, do you just flatten this tensor? I couldn't find the method in the paper

**Comment by Mr81wx** [comment]
average pooling

Kevin Song ***@***.***>于2024年9月20日 周五16:27写道：

**Comment by kvnsng** [comment]
Gotcha. Do you mind sharing the parameters you used for average pooling as well? kernel_size, stride?

**Comment by Mr81wx** [comment]
we just take mean() at the 1st and 2nd dimensions，kernel size 5*121.  There
may be better parameters here.


Kevin Song ***@***.***>于2024年9月20日 周五16:42写道：

**Comment by kvnsng** [comment]
Got it. Thank you!. Just to confirm, this results in embedding size of [256]? I just did torch.mean(vector, dim=[0,1])

**Comment by Mr81wx** [comment]
Hi Kevin,

I saw that your team just launched the new analytics platform with NBA —
congratulations! It looks fantastic and marks an exciting step forward for
basketball analytics. It’s really inspiring to see how your group continues
to bridge advanced modeling with practical applications.

I wanted to reach out to ask whether your team might be interested in
exploring new methods for generating complementary metrics beyond the
existing EPV framework.

I’m currently developing an RL-based “EPV-Plus” model that decomposes
possession value at the individual player level — quantifying both on-ball
and off-ball contributions.

If this direction aligns with your team’s work, I'd be very eager to
explore potential collaboration opportunities.

Best regards,
XING

Kevin Song ***@***.***> 于2024年9月20日周五 23:08写道：

> Got it. Thank you!
>

**Comment by kvnsng** [comment]
Hey Xing - thanks so much. We're very excited about the new partnership. `HoopTransformer` has been a great source of learning for us :) I'd be curious to learn more about the RL-based methods. Feel free to send me an email at songkevinc@gmail.com and we can schedule something?
